azure-iot-hub/azure/iot/hub/iothub_registry_manager.py

```diff
@@ -453,3 +479,424 @@ def delete_module(self, device_id, module_id, etag=None):
+        """
+        input = {
+            "target_condition": target_condition,
+            "custom_metric_queries": custom_metric_queries,
+        }
+        return self.protocol.service.test_configuration_queries(input)
+
+    def bulk_create_or_update_devices(self, devices):
+        """Create, update, or delete the identities of multiple devices from the
+           IoT hub identity registry. Different operations (create, update, delete) on different
+           devices are allowed.
+
+        :param list[ExportImportDevice] devices: The list of device objects to operate on.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: The BulkRegistryOperationResult object.
+        """
+        return self.protocol.service.bulk_create_or_update_devices(devices)
+
+    def query_iot_hub(self, query):
+        """Query an IoT hub to retrieve information regarding device twins using a
+           SQL-like language.
+
+        :param str query: The list of device objects to operate on.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: The BulkRegistryOperationResult object.
+        """
+        query_specification = {"query": query}
+        return self.protocol.service.query_iot_hub(query_specification)
+
+    def apply_configuration_on_edge_device(
+        self, device_id, device_configuration, modules_configuration, module_configuration
+    ):
+        """Applies the provided configuration content to the specified edge
+           device. Modules content is mandantory.
+
+        :param str device_id: The name (Id) of the edge device.
+        :param str device_configuration: The device configuration to apply.
+        :param str modules_configuration: The dictionary of multiple module configurations.
+        :param str module_configuration: The single module configurations.
+
+        :raises: `HttpOperationError<msrest.exceptions.HttpOperationError>`
+            if the HTTP response status is not in [200].
+
+        :returns: An object.
+        """
+        configuration_content = {
+            "device_content": device_configuration,
+            "modules_content": modules_configuration,
+            "module_content": module_configuration,
+        }
+        return self.protocol.service.apply_configuration_on_edge_device(
+            device_id, configuration_content
+        )
+
+    def create_import_export_job(
+        self,
+        job_id,
+        start_time,
+        end_time,
+        job_type,
+        status,
+        progress,
+        input_blob_container_uri,
+        input_blob_name,
+        output_blob_container_uri,
+        output_blob_name,
+        exclude_keys_in_export,
+        failure_reason,
+    ):
+        """Creates a new import/export job on an IoT hub.
```

Comment:
>Creates a new import/export job on an IoT hub. [](start = 11, length = 46)

Is import export job very self evident ? 
Coz I am not sure what this means.

Reply:
No, it is not. I will add a link to the documentation of it.

---
In reply to: [349340093](https://github.com/Azure/azure-iot-sdk-python/pull/397#discussion_r349340093) [](ancestors = 349340093)